daemon/checkpoint.go

```diff
@@ -17,6 +18,13 @@ var (
 	validCheckpointNamePattern = names.RestrictedNamePattern
 )
 
+// checkPoint represents the details of a checkpoint
+type checkPoint struct {
```

Comment:
Why do you not use types.Checkpoint?
https://github.com/moby/moby/blob/master/api/types/types.go#L502
https://github.com/moby/moby/blob/master/daemon/checkpoint.go#L135

Reply:
Because we need to add some fields for types.Checkpoint. And AkihiroSuda tell me I should define a new struct, I accept his suggestion. Please see the first review msg at the top of this page. Thanks.